_docs/koppelaal-1.2/write-connector.md

```diff
@@ -0,0 +1,125 @@
+---
+title: Write a Koppeltaal Connector
+category: How-to
+order: 2
+---
+
+# How-to: Koppeltaal connector
+
+## About this guide
+This guide will not go into Koppeltaal terminology, but rather assumes this knowledge is available to the reader. Whether it is from memory or the [Technical Design Document]. This document is not technical, but rather a descriptive / explanatory guide to assist the implementer towards a usable connector. Mainly by answering question in advance, that we think developers will bump into.
+
+## Connector repository and source code
+Since the connector will be used by other parties and will also be handed over to the Koppeltaal Community at some point, it’s important that the source code is clear and readable for other developers that want to debug or add features in the (near) future.
+
+A git repository can be created at the [Koppeltaal organization on GitHub]. Sources of connectors in other programming languages can be viewed there as well.
+
+## Where to start
+- Find out which FHIR resources need to be communicated about. Application dependent.
+- Is there FHIR model library in my programming language?
+
+## Connector responsibilities
+A Koppeltaal connector is a library to be used by an application written in particular programming language. It communicates directly with the Koppeltaal Server REST interface, while exposing a simple API to the application, exchanging custom objects. Koppeltaal specific complexity should be handled by the connector - e.g., composing a message that conforms to Koppeltaal ontology.
+
+_Note_: The Koppeltaal server supports both JSON and XML.
+
+![figure 1]
+![placeholder]
+**Figure 1**. _Each application communicates with the Koppeltaal server through a connector to encapsulate most of the complexity._
+
+Obviously, the functionalities a connector should provide depends on the applications that use it. However, we think there is a set of basic functionalities that a connector in general should support.
+- Retrieve the `Conformance Statement`
+- Send/retrieve `Messages` to/from a domain
+-- Create a Message bundle with FHIR Resources (send)
+-- Extract FHIR Resources from the Message bundle (receive)
+-- Update the _ProcessingStatus_ of a Message
+- Retrieve and update `Activity Definitions`
+
+Additionally, a connector could also support the following functionality:
+- Create a RESTful endpoint and configure a webhook on KT server
```

Comment:
Unfortunately, I havent't found documentation on this yet. When it's there I will definitely link to that.